http-client-jdk/src/test/groovy/io/micronaut/http/client/jdk/ClientProxySpec.groovy

```diff
@@ -0,0 +1,127 @@
+                    .withExposedPorts(PROXY_PORT)
+                    .withLogConsumer { outputFrame -> print("SQUID::\t${outputFrame.getUtf8String()}") }
+                    .waitingFor(new HostPortWaitStrategy())
+
+    @AutoCleanup
+    EmbeddedServer embeddedServer
+
+    def setup() {
+        proxyContainer.start()
+    }
+
+    void "test downloading via http proxy using proxy-address"() {
+        given:
+        startServer([
+                'micronaut.http.client.exception-on-error-status': false,
+                'micronaut.http.client.proxy-type'               : 'http',
+                'micronaut.http.client.proxy-address'            : "${proxyHost}:${proxyPort}"
+        ])
+
+        when: 'download page via proxy'
+        def response = downloadPage()
+
+        then: 'page is downloaded'
+        response.status == HttpStatus.OK
+        response.body().contains(HTML_FRAGMENT)
+
+        when: 'proxy container is stopped'
+        proxyContainer.stop()
+
+        then: 'page download fails'
+        downloadFailsWithException() instanceof HttpClientException
+    }
+
+    void "test downloading via http proxy using default proxy-selector"() {
+        given:
+        def oldProxyHost = System.setProperty('https.proxyHost', proxyHost)
+        def oldProxyPort = System.setProperty('https.proxyPort', proxyPort.toString())
+        startServer([
+                'micronaut.http.client.exception-on-error-status': false,
+                'micronaut.http.client.proxy-selector'           : 'default'
+        ])
+
+        when: 'download page via proxy'
+        def response = downloadPage()
+
+        then: 'page is downloaded'
+        response.status == HttpStatus.OK
+        response.body().contains(HTML_FRAGMENT)
+
+        when: 'proxy container is stopped'
+        proxyContainer.stop()
+
+        then: 'page download fails'
+        downloadFailsWithException() instanceof HttpClientException
+
+        cleanup:
+        resetSystemProperty('https.proxyHost', oldProxyHost)
+        resetSystemProperty('https.proxyPort', oldProxyPort)
+    }
+
+    private String getProxyHost() {
+        proxyContainer.host
+    }
+
+    private int getProxyPort() {
+        def port = proxyContainer.getMappedPort(PROXY_PORT)
+        println("Proxy port: ${port}")
+        port
+    }
+
+    private void startServer(Map<String, Object> config) {
+        embeddedServer = ApplicationContext.run(EmbeddedServer, config, Environment.TEST)
+    }
+
+    private HttpClient getClient() {
+        embeddedServer.applicationContext.getBean(HttpClient)
+    }
+
+    private HttpResponse<String> downloadPage() {
+        client.toBlocking().exchange(URL, String)
+    }
+
+    private Throwable downloadFailsWithException() {
+        try {
+            downloadPage()
+            return null
+        } catch (Throwable ex) {
+            return ex
+        }
+    }
+
+    private static void resetSystemProperty(String property, String value) {
```

Comment:
isnt there a spock feature for this

Reply:
Good catch https://github.com/micronaut-projects/micronaut-core/pull/8441/commits/32e5fa22aec2b6565dc0248a2d0989f0f1deda4e